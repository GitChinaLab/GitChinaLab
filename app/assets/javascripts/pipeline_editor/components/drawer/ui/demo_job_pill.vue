<script>
export default {
  props: {
    jobName: {
      type: String,
      required: true,
    },
  },
};
</script>
<template>
  <div
    class="gl-w-13 gl-h-6 gl-font-sm gl-bg-white gl-inset-border-1-blue-500 gl-text-center gl-text-truncate gl-rounded-pill gl-px-4 gl-py-2 gl-relative gl-z-index-1 gl-transition-duration-slow gl-transition-timing-function-ease"
  >
    {{ jobName }}
  </div>
</template>
