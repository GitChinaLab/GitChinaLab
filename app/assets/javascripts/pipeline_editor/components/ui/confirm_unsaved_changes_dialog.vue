<script>
export default {
  props: {
    hasUnsavedChanges: {
      type: Boolean,
      required: true,
    },
  },
  created() {
    window.addEventListener('beforeunload', this.confirmChanges);
  },
  destroyed() {
    window.removeEventListener('beforeunload', this.confirmChanges);
  },
  methods: {
    confirmChanges(e = {}) {
      if (this.hasUnsavedChanges) {
        e.preventDefault();
        // eslint-disable-next-line no-param-reassign
        e.returnValue = ''; // Chrome requires returnValue to be set
      }
    },
  },
  render: () => null,
};
</script>
