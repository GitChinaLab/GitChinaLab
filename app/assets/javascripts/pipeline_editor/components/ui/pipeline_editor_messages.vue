<script>
import { GlAlert } from '@gitlab/ui';
import { getParameterValues, removeParams } from '~/lib/utils/url_utility';
import { __, s__ } from '~/locale';
import {
  COMMIT_FAILURE,
  COMMIT_SUCCESS,
  DEFAULT_FAILURE,
  DEFAULT_SUCCESS,
  LOAD_FAILURE_UNKNOWN,
  PIPELINE_FAILURE,
} from '../../constants';
import CodeSnippetAlert from '../code_snippet_alert/code_snippet_alert.vue';
import {
  CODE_SNIPPET_SOURCE_URL_PARAM,
  CODE_SNIPPET_SOURCES,
} from '../code_snippet_alert/constants';

export default {
  components: {
    GlAlert,
    CodeSnippetAlert,
  },
  errorTexts: {
    [COMMIT_FAILURE]: s__('Pipelines|The GitLab CI configuration could not be updated.'),
    [DEFAULT_FAILURE]: __('Something went wrong on our end.'),
    [LOAD_FAILURE_UNKNOWN]: s__('Pipelines|The CI configuration was not loaded, please try again.'),
    [PIPELINE_FAILURE]: s__('Pipelines|There was a problem with loading the pipeline data.'),
  },
  successTexts: {
    [COMMIT_SUCCESS]: __('Your changes have been successfully committed.'),
    [DEFAULT_SUCCESS]: __('Your action succeeded.'),
  },
  props: {
    failureType: {
      type: String,
      required: false,
      default: null,
    },
    failureReasons: {
      type: Array,
      required: false,
      default: () => [],
    },
    showFailure: {
      type: Boolean,
      required: false,
      default: false,
    },
    showSuccess: {
      type: Boolean,
      required: false,
      default: false,
    },
    successType: {
      type: String,
      required: false,
      default: null,
    },
  },
  data() {
    return {
      codeSnippetCopiedFrom: '',
    };
  },
  computed: {
    failure() {
      switch (this.failureType) {
        case LOAD_FAILURE_UNKNOWN:
          return {
            text: this.$options.errorTexts[LOAD_FAILURE_UNKNOWN],
            variant: 'danger',
          };
        case COMMIT_FAILURE:
          return {
            text: this.$options.errorTexts[COMMIT_FAILURE],
            variant: 'danger',
          };
        case PIPELINE_FAILURE:
          return {
            text: this.$options.errorTexts[PIPELINE_FAILURE],
            variant: 'danger',
          };
        default:
          return {
            text: this.$options.errorTexts[DEFAULT_FAILURE],
            variant: 'danger',
          };
      }
    },
    success() {
      switch (this.successType) {
        case COMMIT_SUCCESS:
          return {
            text: this.$options.successTexts[COMMIT_SUCCESS],
            variant: 'info',
          };
        default:
          return {
            text: this.$options.successTexts[DEFAULT_SUCCESS],
            variant: 'info',
          };
      }
    },
  },
  created() {
    this.parseCodeSnippetSourceParam();
  },
  methods: {
    dismissCodeSnippetAlert() {
      this.codeSnippetCopiedFrom = '';
    },
    dismissFailure() {
      this.$emit('hide-failure');
    },
    dismissSuccess() {
      this.$emit('hide-success');
    },
    parseCodeSnippetSourceParam() {
      const [codeSnippetCopiedFrom] = getParameterValues(CODE_SNIPPET_SOURCE_URL_PARAM);
      if (codeSnippetCopiedFrom && CODE_SNIPPET_SOURCES.includes(codeSnippetCopiedFrom)) {
        this.codeSnippetCopiedFrom = codeSnippetCopiedFrom;
        window.history.replaceState(
          {},
          document.title,
          removeParams([CODE_SNIPPET_SOURCE_URL_PARAM]),
        );
      }
    },
  },
};
</script>

<template>
  <div>
    <code-snippet-alert
      v-if="codeSnippetCopiedFrom"
      :source="codeSnippetCopiedFrom"
      class="gl-mb-5"
      @dismiss="dismissCodeSnippetAlert"
    />
    <gl-alert
      v-if="showSuccess"
      :variant="success.variant"
      class="gl-mb-5"
      @dismiss="dismissSuccess"
    >
      {{ success.text }}
    </gl-alert>
    <gl-alert
      v-if="showFailure"
      :variant="failure.variant"
      class="gl-mb-5"
      @dismiss="dismissFailure"
    >
      {{ failure.text }}
      <ul v-if="failureReasons.length" class="gl-mb-0">
        <li v-for="reason in failureReasons" :key="reason">{{ reason }}</li>
      </ul>
    </gl-alert>
  </div>
</template>
