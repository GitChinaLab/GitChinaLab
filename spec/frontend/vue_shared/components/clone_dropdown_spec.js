import { GlFormInputGroup, GlDropdownSectionHeader } from '@gitlab/ui';
import { shallowMount } from '@vue/test-utils';
import CloneDropdown from '~/vue_shared/components/clone_dropdown.vue';

describe('Clone Dropdown Button', () => {
  let wrapper;
  const sshLink = 'ssh://foo.bar';
  const httpLink = 'http://foo.bar';
  const httpsLink = 'https://foo.bar';
  const defaultPropsData = {
    sshLink,
    httpLink,
  };

  const createComponent = (propsData = defaultPropsData) => {
    wrapper = shallowMount(CloneDropdown, {
      propsData,
      stubs: {
        'gl-form-input-group': GlFormInputGroup,
      },
    });
  };

  afterEach(() => {
    wrapper.destroy();
    wrapper = null;
  });

  describe('rendering', () => {
    it('matches the snapshot', () => {
      createComponent();
      expect(wrapper.element).toMatchSnapshot();
    });

    it.each`
      name      | index | value
      ${'SSH'}  | ${0}  | ${sshLink}
      ${'HTTP'} | ${1}  | ${httpLink}
    `('renders correct link and a copy-button for $name', ({ index, value }) => {
      createComponent();
      const group = wrapper.findAll(GlFormInputGroup).at(index);
      expect(group.props('value')).toBe(value);
      expect(group.find(GlFormInputGroup).exists()).toBe(true);
    });

    it.each`
      name          | value
      ${'sshLink'}  | ${sshLink}
      ${'httpLink'} | ${httpLink}
    `('does not fail if only $name is set', ({ name, value }) => {
      createComponent({ [name]: value });

      expect(wrapper.find(GlFormInputGroup).props('value')).toBe(value);
      expect(wrapper.findAll(GlDropdownSectionHeader).length).toBe(1);
    });
  });

  describe('functionality', () => {
    it.each`
      name          | value
      ${'sshLink'}  | ${null}
      ${'httpLink'} | ${null}
    `('allows null values for the props', ({ name, value }) => {
      createComponent({ ...defaultPropsData, [name]: value });

      expect(wrapper.findAll(GlDropdownSectionHeader).length).toBe(1);
    });

    it('correctly calculates httpLabel for HTTPS protocol', () => {
      createComponent({ httpLink: httpsLink });
      expect(wrapper.find(GlDropdownSectionHeader).text()).toContain('HTTPS');
    });
  });
});
