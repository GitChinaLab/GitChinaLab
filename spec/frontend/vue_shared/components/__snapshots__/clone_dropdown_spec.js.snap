// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clone Dropdown Button rendering matches the snapshot 1`] = `
<gl-dropdown-stub
  category="primary"
  clearalltext="Clear all"
  clearalltextclass="gl-px-5"
  headertext=""
  hideheaderborder="true"
  highlighteditemstitle="Selected"
  highlighteditemstitleclass="gl-px-5"
  right="true"
  size="medium"
  text="Clone"
  variant="info"
>
  <div
    class="pb-2 mx-1"
  >
    <gl-dropdown-section-header-stub>
      Clone with SSH
    </gl-dropdown-section-header-stub>
     
    <div
      class="mx-3"
    >
      <div
        readonly="readonly"
      >
        <b-input-group-stub
          tag="div"
        >
          <!---->
           
          <b-form-input-stub
            class="gl-form-input"
            debounce="0"
            formatter="[Function]"
            readonly="true"
            type="text"
            value="ssh://foo.bar"
          />
           
          <b-input-group-append-stub
            tag="div"
          >
            <gl-button-stub
              aria-label="Copy URL"
              buttontextclasses=""
              category="primary"
              class="d-inline-flex"
              data-clipboard-text="ssh://foo.bar"
              data-qa-selector="copy_ssh_url_button"
              icon="copy-to-clipboard"
              size="medium"
              title="Copy URL"
              variant="default"
            />
          </b-input-group-append-stub>
        </b-input-group-stub>
      </div>
    </div>
     
    <gl-dropdown-section-header-stub>
      Clone with HTTP
    </gl-dropdown-section-header-stub>
     
    <div
      class="mx-3"
    >
      <div
        readonly="readonly"
      >
        <b-input-group-stub
          tag="div"
        >
          <!---->
           
          <b-form-input-stub
            class="gl-form-input"
            debounce="0"
            formatter="[Function]"
            readonly="true"
            type="text"
            value="http://foo.bar"
          />
           
          <b-input-group-append-stub
            tag="div"
          >
            <gl-button-stub
              aria-label="Copy URL"
              buttontextclasses=""
              category="primary"
              class="d-inline-flex"
              data-clipboard-text="http://foo.bar"
              data-qa-selector="copy_http_url_button"
              icon="copy-to-clipboard"
              size="medium"
              title="Copy URL"
              variant="default"
            />
          </b-input-group-append-stub>
        </b-input-group-stub>
      </div>
    </div>
  </div>
</gl-dropdown-stub>
`;
